modules/bpi/bpi.module

```diff
@@ -148,46 +149,90 @@ function bpi_menu() {
     'file' => 'bpi.admin.inc',
   );
 
+  $menu['bpi/edit/%'] = array(
+    'title' => 'BPI content edit',
+    'access callback' => 'bpi_access',
+    'access arguments' => ['bpi content edit', 2],
+    'page callback' => '_bpi_edit_callback',
+    'page arguments' => array(2),
+    'type' => MENU_CALLBACK,
+  );
+
   return $menu;
 }
 
 /**
- * Check if current user has access to the BPI Syndication menu.
+ * Redirects the user to BPI node edit form.
+ *
+ * @param string $bid
+ *   BPI id.
+ */
+function _bpi_edit_callback($bid) {
+  if (!$node = bpi_load_node_by_bid($bid)) {
+    drupal_not_found();
+  }
+
+  $path = 'node/' . $node->nid . '/edit';
+  drupal_goto($path);
+}
+
+/**
+ * Custom access checker for BPI actions.
+ *
+ * BPI overview:
+ * The logged in user must both have the
+ * "bpi syndicate content" permission and access to create content from BPI.
+ * BPI edit link:
+ * The logged in user must have access to the target node. This is the place
+ * where BPI id parameter is used.
  *
- * The logged in user must both have the "bpi syndicate content" permission
- * and access to create content from BPI.
+ * @param string $access_permission
+ *   Permission identifier.
+ * @param string $bid
+ *   Optional BPI article id.
  *
  * @return bool
  *   TRUE on access, FALSE otherwise
  */
-function bpi_access() {
+function bpi_access($access_permission, $bid = null) {
   global $user;
-  // Disallow if user doesn't have permission to syndicate bpi content.
-  if (!user_access('bpi syndicate content')) {
-    return FALSE;
-  }
 
-  // Get roles that are allowed to create content from BPI according to
-  // Workflow settings.
-  $record = NULL;
-  try {
-    $query = db_select('workflow_transitions', 'w');
-    $query->fields('w');
-    $query->condition('w.name', '_creation_to_created_bpi');
-    $record = $query->execute()->fetchAssoc();
-  }
-  catch (PDOException $e) {
-    bpi_error_message($e, 'error');
-  }
+  $access_result = FALSE;
+
+  switch ($access_permission) {
+    case 'bpi syndicate content':
+      // Get roles that are allowed to create content from BPI according to
+      // Workflow settings.
+      $record = NULL;
+      try {
+        $query = db_select('workflow_transitions', 'w');
+        $query->fields('w');
+        $query->condition('w.name', '_creation_to_created_bpi');
+        $record = $query->execute()->fetchAssoc();
+      }
```

Comment:
Have you considered using the API provided by the Workflow module to fetch this data? [`Workflow::getTransitions()`](https://git.drupalcode.org/project/workflow/-/blob/7.x-2.x/includes/Entity/WorkflowInterface.php#L168) seems relevant.

Reply:
Well, this was not added in this PR.
But, answering this, I found Workflow methods unrealiable, as it would turn out into such construct:
```
/** @var \Workflow $workflow */
  $workflow = entity_load_single('Workflow', 'bpi');
  /** @var \WorkflowTransition[] $transitions */
  $transitions = $workflow->getTransitions();
  foreach ($transitions as $transition) {
    if ('_creation_to_created_bpi' === $transition->name) {
      $record = $transition;
      break;
    }
  }
```

Reply:
Thanks for explaining.